src/overlays/actors/ovl_Arrow_Ice/z_arrow_ice.c

```diff
@@ -21,21 +33,214 @@ const ActorInit Arrow_Ice_InitVars = {
     (ActorFunc)ArrowIce_Update,
     (ActorFunc)ArrowIce_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Arrow_Ice/func_80867CD0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Arrow_Ice/ArrowIce_Init.s")
+static InitChainEntry initChain[] = {
+    ICHAIN_F32(unk_F4, 2000, ICHAIN_STOP),
+};
+
+void ArrowIce_SetupAction(ArrowIce* this, ActorFunc* actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void ArrowIce_Init(ArrowIce* this, GlobalContext* globalCtx) {
+    Actor_ProcessInitChain(&this->actor, initChain);
+    this->radius = 0;
+    this->unk_160 = 1.0f;
+    ArrowIce_SetupAction(this, ArrowIce_Charge);
+    Actor_SetScale(&this->actor, 0.01);
```

Comment:
Minor consistency update
```suggestion
    Actor_SetScale(&this->actor, 0.01f);
```

Reply:
oh good catch, thanks